### ISSUE TITLE
[testing-coverage]: Expand test coverage to `site/assets/app.mjs`

### ISSUE THREAD

**Issue by davidlabianca** [cosai-oasis/secure-ai-tooling#224]
### Enhancement Title*

[testing-coverage]: Expand test coverage to `site/assets/app.mjs`

### Enhancement Category*

Tooling / Scripts / Validation

### Overview*

`site/assets/app.mjs` is the persona-pages DOM renderer and has zero direct test coverage. All functions are module-private (no `export`), making them unreachable from `node --test` without either a refactor or adding a DOM emulator. The Node-test conventions at `site/tests/README.md` explicitly scope current tests to pure-logic modules and defer DOM-testing decisions to a follow-up PR — this issue is that follow-up.

### Rationale*

The absence of tests means regressions in any of the following are only caught by manual inspection:

- HTML escaping (`escapeHtml`) — a hole here becomes an XSS vector.
- Template-literal interpolation into `innerHTML` (all `renderXxx` functions).
- Post-[BLOCK-02] subsection rendering in `renderRichParagraphs` — the array-vs-string branch that wraps nested prose in `<div class="subsection">`.
- Conditional rendering branches (empty states, guided-vs-manual persona flows, governance-only results).
- Event-handler state transitions (answer changes, manual selection toggles, persona overrides, reset).

## Current uncovered surface

By category (line ranges approximate, refer to `app.mjs` on post-rebase HEAD):

| Category                            | Examples                                                                                                                 | Testability                                                 |
| ----------------------------------- | ------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------------------- |
| Pure HTML-string builders           | `escapeHtml`, `renderRichParagraphs`, `renderStepRail`, `renderPersonaBadges`, `renderRiskGroups`, `renderControlGroups` | **Trivial** once exported                                   |
| Composite renderers with state read | `renderIntroduction`, `renderQuestionGroups`, `renderSummary`, `renderResults`                                           | **Trivial** once exported (state is plain object passed in) |
| DOM writers                         | `renderApp` (single entry point that sets `appElement.innerHTML`)                                                        | Requires DOM or string-capture                              |
| Event handlers + state mutation     | Two `appElement.addEventListener` blocks                                                                                 | Requires DOM                                                |
| Data fetch                          | `loadSiteData` (one `fetch` call)                                                                                        | Requires fetch mock                                         |

### Scope*

Medium (multiple files, moderate complexity)

### Breaking Changes*

No - Fully backward compatible

## Recommended path — Option A: refactor for direct testability

**Extract pure HTML-string builders into a new module** `site/assets/app-render.mjs`. Every function that currently consumes `state` and returns an HTML string (or fragment) becomes an exported pure function in the new module. `app.mjs` retains only DOM writes, event wiring, and state management.

After refactor:

```
site/assets/
├── persona-logic.mjs      # pure logic (today)
├── app-render.mjs         # NEW — pure HTML string builders (exported)
└── app.mjs                # DOM mutation + event wiring (thin)
```

**Coverage plan:** add `site/tests/app-render.test.mjs` following conventions in `site/tests/README.md`. Tests cover:

- Each builder's output shape against hand-built state fixtures.
- `escapeHtml` against `<script>`, `"`, `'`, `&`, `<`, `>` inputs.
- `renderRichParagraphs` string-vs-array-vs-empty inputs (locks in the BLOCK-02 subsection contract).
- Conditional branches: zero personas matched, governance-only persona (no risks), all personas manually deselected.
- Category-grouping edge cases against `renderRiskGroups` / `renderControlGroups`.

**No new dependencies.** Matches the zero-dep property codified in `site/tests/README.md`.

**Scope:** ~200 lines of refactor in `app.mjs`; ~120 lines of new tests; no visual change to the shipped site.

## Alternate path — Option B: DOM integration tests with jsdom

If Option A leaves material gaps (event handlers, full render pipeline, state machine), file a **separate** follow-up per the Escalation process in `site/tests/README.md`:

1. Justify the need with concrete integration cases not expressible under Option A.
2. Pin `jsdom` in `package.json`.
3. Run DOM tests with `--test-concurrency=1` to serialize shared-DOM mutation.
4. Update `site/tests/README.md` with new conventions.
5. Update `.github/workflows/persona-pages.yml` to run the new test set.

Option B is not a prerequisite for closing this issue; most value is captured by Option A.

## Acceptance criteria

- [ ] Every pure HTML-string builder in `app.mjs` has a direct test in `app-render.test.mjs`.
- [ ] [BLOCK-02] subsection rendering (`<div class="subsection">`) is tested against string, array, and mixed inputs.
- [ ] `escapeHtml` is tested against all five HTML entity classes plus script-injection-adjacent strings.
- [ ] New test suite runs under `node --test site/tests/*.test.mjs` with zero new dependencies.
- [ ] No visual change to the deployed site verified by manual spot-check.
- [ ] `site/tests/README.md` updated if any convention shifts are introduced.
- [ ] **Focus management after DOM patches.** The current `renderApp()` implementation wipes `<main data-app>` via `innerHTML` on every state change, which destroys the focused node and drops focus to `<body>`. After this refactor: focus is preserved across state transitions (answering a question, toggling a manual persona, switching step tabs). Extends the `aria-live` status region added by PR #223 [NIT-12] layer 1.

## Scope estimate

| Path                | Effort                    | Risk                                               |
| ------------------- | ------------------------- | -------------------------------------------------- |
| Option A            | ~1 day (refactor + tests) | Low — no visual change, no dep change              |
| Option A + Option B | ~3 days                   | Medium — Escalation PR, DOM concurrency, CI tuning |

Recommend starting with Option A; re-evaluate need for Option B after it lands.

## Dependencies

- **Blocked by:** merging of PR #223 (to avoid rebase conflicts with [BLOCK-02] edits to `renderRichParagraphs`).
- **Should precede:** any further UI feature work on `app.mjs` so new features can be tested from day one.

## Out of scope

- Visual-regression testing (separate tool, separate issue).
- Cross-browser end-to-end (Playwright/Cypress — separate issue).
- Additional accessibility work beyond focus-management: `:focus-visible` and `prefers-reduced-motion` land in PR #223 via [REC-19]/[REC-20]; `aria-live` layer 1 lands in PR #223 via [NIT-12]; keyboard-shortcut discoverability, high-contrast mode, and screen-reader-specific testing are tracked separately.

**Comment by zeal-somani** [comment]
Great first MVP. Additional thoughts

AI-Specific SBOM Standards: There is a lack of a standardized [Software Bill of Materials (SBOM)](https://www.coalitionforsecureai.org/addressing-whats-next-in-securing-enterprise-ai/) that captures model provenance, training data lineage, and fine-tuning history.

Inference-Based Leakage Controls: Traditional pattern matching (DLP) fails to catch data leakage occurring via [inference, paraphrasing, or summary](https://www.coalitionforsecureai.org/when-the-bots-run-the-incident-response-what-ai-agents-mean-for-enterprise-security/). The framework currently lacks proven mitigations for this "semantic mosaic effect."

Agentic Permission Boundaries related Controls: Current maps do not fully account for the risk where [natural language instructions](https://www.coalitionforsecureai.org/addressing-whats-next-in-securing-enterprise-ai/) can modify an agent's own configuration or bypass restricted shell commands.

Executable Intent in Non-Code Files: There is a gap in scanning artifacts like .pt or .bin files that carry executable intent but sit outside traditional SAST/DAST scanner definitions.

**Comment by zeal-somani** [comment]
Proposal: Implementing Progressive Disclosure for Outcome-Oriented UX
To eliminate the "0 Risk Results" dead-end and reduce upfront cognitive load, I suggest transitioning the persona selection flow to a guided, project-based model.

Step 1: Project-Based Intake Questions
Instead of requiring users to self-identify as an "Auditor" or "Integrator" immediately, the interface should present 3–5 high-level questions about their project. For example, the tool could ask if they are building a model from scratch versus using a third-party API, or if the system will handle sensitive data like PII or clinical records.

Step 2: Intelligent Background Mapping
As the user provides these answers, the application logic should automatically map their project profile to relevant personas in the background. For instance, selecting "3rd-party API" and "PII" would automatically flag both the Integrator and Compliance Auditor personas without requiring manual toggling.

Step 3: The "Bridge" Transition
Once the intake is complete, the tool should provide a clear summary to transition the user to their results. A prompt could state: "Based on your project profile, we’ve identified risks relevant to an Integrator and Auditor. Would you like to refine these roles further or jump straight to your risk report?"
